lib/stats/tests/test_stats_cluster.c

```diff
@@ -163,6 +163,61 @@ Test(stats_cluster, test_stats_cluster_key_equal_when_custom_tags_are_the_same)
   stats_cluster_free(sc2);
 }
 
+Test(stats_cluster, test_stats_cluster_key_equal_with_labels)
+{
+  StatsClusterLabel labels[] =
+  {
+    stats_cluster_label("app", "cisco"),
+    stats_cluster_label("sourceip", "127.0.0.1"),
+  };
+  StatsClusterLabel labels2[] = { stats_cluster_label("app", "cisco") };
+
+  StatsClusterKey sc_key1;
+  StatsClusterKey sc_key2;
+  stats_cluster_logpipe_key_set(&sc_key1, "id", labels, G_N_ELEMENTS(labels));
+  stats_cluster_logpipe_key_set(&sc_key2, "id", labels, G_N_ELEMENTS(labels));
+  StatsCluster *sc1 = stats_cluster_new(&sc_key1);
+  StatsCluster *sc2 = stats_cluster_new(&sc_key2);
+
+  cr_assert(stats_cluster_key_equal(&sc_key1, &sc_key2), "%s", __FUNCTION__);
+
+  stats_cluster_free(sc2);
+
+  stats_cluster_logpipe_key_set(&sc_key2, "id", labels2, G_N_ELEMENTS(labels2));
+  sc2 = stats_cluster_new(&sc_key2);
+
+  cr_assert_not(stats_cluster_key_equal(&sc_key1, &sc_key2), "%s", __FUNCTION__);
+
+  stats_cluster_free(sc2);
+  stats_cluster_free(sc1);
+}
+
+Test(stats_cluster, test_stats_cluster_key)
```

Comment:
I was thinking about the comparisons. If I understand correctly there are these variables:
 1. Keys cannot be NULLs, so they can be:
     a. same (non-NULL, non-NULL)
     b. different (non-NULL, non-NULL)
 2. Labels can be NULLs, so they can be:
     a. same (NULL, NULL),
     b. same (non-NULL, non-NULL)
     c. different (NULL, non-NULL)
     d. different (non-NULL, non-NULL)

From these, we can create 2x4=8 test scenarios. I have made a table for them with examples (comparing A and B):
| 1. keys      | key-A | key-B | 2. labels                         | label-A | label-B | result |
| ------------ | ----- | ----- | --------------------------------- | ------- | ------- | ------ |
| a. same      | key-1 | key-1 | a. same (NULL, NULL)              | NULL    | NULL    | ==     |
| a. same      | key-1 | key-1 | b. same (non-NULL, non-NULL)      | label-1 | label-1 | ==     |
| a. same      | key-1 | key-1 | c. different (NULL, non-NULL)     | NULL    | label-1 | !=     |
| a. same      | key-1 | key-1 | d. different (non-NULL, non-NULL) | label-1 | label-2 | !=     |
| b. different | key-1 | key-2 | a. same (NULL, NULL)              | NULL    | NULL    | !=     |
| b. different | key-1 | key-2 | b. same (non-NULL, non-NULL)      | label-1 | label-1 | !=     |
| b. different | key-1 | key-2 | c. different (NULL, non-NULL)     | NULL    | label-1 | !=     |
| b. different | key-1 | key-2 | d. different (non-NULL, non-NULL) | label-1 | label-2 | !=     |

Some of them are not tested here, and I think they should be.

Reply:
Done, thanks for the table.